aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1557,7 +1557,7 @@ inline void baddbmm_cpu_kernel(const Tensor& result, const Tensor& self, const T
                 r += s2[k] * m1[k][j];
               }
             } else {
-              r *= beta;
+              r = beta == scalar_t(0) ? scalar_t(0) : beta * r;
               for (const auto k : c10::irange(ks)) {
```

Comment:
Oh, wow, I did not know that `baddmm` has an in-house implementation...

Reply:
I was sure matmuls are delegated to MKL or something of the sort.
This bug seems to have been there for the past 5 years, I'm amazed there haven't been reported random failures of deployed CPU inference systems.

Reply:
haha, maybe that's why :) everyone who cares about inference, made sure to use MKL and bypass this buggy impl :)

Reply:
Yeah, unless this path bypasses MKL when it is not available, I would assume dispatching to `aten::mm` could be a better strategy. If that is the case indeed, this could be done as a follow-up?

Reply:
@nikitaved Never tried, `torch.mm` was never powerful enough for my purposes (no broadcast, single matrix). Not for real world heavy lifting.

Reply:
It does broadcast, however:
```python
In [1]: import torch

In [2]: x = torch.rand(3, 3)

In [3]: y = torch.rand(3, 3, 3, 3)

In [4]: (x @ y).shape
Out[4]: torch.Size([3, 3, 3, 3])

```

Reply:
According to the docs, that's not `torch.mm`. Probably `torch.matmul`. 
I never figured why there are so many entrypoints.

Reply:
Yeah, it is `matmul` indeed. `mm` appears to be a method. Totally agree, seems this redundancy is more confusing than it is helpful...

Reply:
baddbmm is usually dispatched to mkl, but in some cases (small matrices, small batches) it goes to special implementation).